lisa/analysis/base.py

```diff
@@ -152,16 +152,29 @@ def save_plot(self, figure, filepath=None, img_format=None):
         default_dir = '.'
         return self._save_plot(figure, default_dir, filepath, img_format)
 
-    def _do_plot(self, plotter, filepath, axis):
+    def do_plot(self, plotter, filepath, axis, **kwargs):
         """
         Simple helper for consistent behavior across methods.
+
+        :returns: An :class:`matplotlib.axes.Axes` containing the plot.
+
+        :param filepath: Path of the file to save the figure in. If `None`, no
+            file is saved.
+        :type filepath: str or None
+
+        :param axis: instance of :class:`matplotlib.axes.Axes` to plot into.
+            If `None`, a new figure and axis are created and returned.
+        :type axis: matplotlib.axes.Axes or None
```

Comment:
That can still be `numpy.ndarray(matplotlib.axes.Axes)` for plots that use multiple `Axes` (e.g. `frequency.plot_domain_frequencies`), right?

Reply:
yep, forgot about that case